### PR TITLE
Sc 241 write integration test simulating a token rescue on

### PR DESCRIPTION
**Pull request by FHieser**
I created a E2E Test that shows the steps needed for a BondingCurve FundingManager Transferal.
Should I add more tests, that check if the transferal worked? If so which tests would you like to see?

### CODE REVIEW

File: test/e2e/fundingManager/BondingCurveTokenRescueE2E.t.sol [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+        // IMPORTANT
+        // =========
+        // Due to how the underlying rebase mechanism works, it is necessary
+        // to always have some amount of tokens in the fundingmanager.
+        // It's best, if the owner deposits them right after deployment.
+        // uint initialDeposit = 10e18;
+        // token.mint(address(this), initialDeposit);
+        // token.approve(address(fundingManager), initialDeposit);
+        // fundingManager.deposit(initialDeposit);
+
```
Comment (Zitzak) [inline test/e2e/fundingManager/BondingCurveTokenRescueE2E.t.sol:130]: Can be removed, outdated

File: test/e2e/fundingManager/BondingCurveTokenRescueE2E.t.sol [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+            _create_E2E_Orchestrator(workflowConfig, moduleConfigurations);
+
+        FM_BC_Bancor_Redeeming_VirtualSupply_v1 fundingManager =
+        FM_BC_Bancor_Redeeming_VirtualSupply_v1(
+            address(orchestrator.fundingManager())
+        );
+
+        issuanceToken.setMinter(address(fundingManager), true);
+
+        // IMPORTANT
```
Comment (Zitzak) [inline test/e2e/fundingManager/BondingCurveTokenRescueE2E.t.sol]: This section can be removed, outdated

File: test/e2e/fundingManager/BondingCurveTokenRescueE2E.t.sol [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+            assertTrue(issuanceToken.balanceOf(alice) > 0);
+        }
+        vm.stopPrank();
+
+        // --------------------------------------------------------------------------------
+        // Start Token Rescue
+
+        // Stop minting of new tokens
+
+        issuanceToken.setMinter(address(fundingManager), false);
```
Comment (Zitzak) [inline test/e2e/fundingManager/BondingCurveTokenRescueE2E.t.sol:151]: We should also pause the buy/sell operation for the FM here. Of course issuance is not possible anymore but redeeming would still work I believe

File: test/e2e/fundingManager/BondingCurveTokenRescueE2E.t.sol [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+            address(token), // token
+            // This represented the allowed amount of collateral token to be transferred
+            token.balanceOf(address(fundingManager))
+                - fundingManager.projectCollateralFeeCollected(), // amount
+            block.timestamp, // startTime
+            0, // cliff
+            block.timestamp // endTime
+        );
+
+        // Set Minter for Issuance to new BC
```
Comment (Zitzak) [inline test/e2e/fundingManager/BondingCurveTokenRescueE2E.t.sol]:
```suggestion
        // Initiate setting of new Funding Manager
```

**Review by Zitzak — CHANGES_REQUESTED** [review]
Few comment changes, otherwise looks good

File: test/e2e/fundingManager/BondingCurveTokenRescueE2E.t.sol (hunk 2 above)
Comment (FHieser) [reply test/e2e/fundingManager/BondingCurveTokenRescueE2E.t.sol]: You mean the section about how we need a minimum amount of tokens in the contract to start?

Comment (Zitzak) [reply test/e2e/fundingManager/BondingCurveTokenRescueE2E.t.sol]: I mean the section about the rebasing mechanism, which we don't have anymore

**Review by Zitzak — APPROVED** [review]
LGTM

**Review by 0xNuggan — APPROVED** [review]
LGTM!

**Pull request merged**